### PR TITLE
Delete useless in controller and secure some endpoints

### PR DESCRIPTION
**Pull request by MarrMika**

### CODE REVIEW

File: src/main/java/com/ita/if103java/ims/controller/UserController.java [hunk 1]
```diff
@@ -91,12 +79,13 @@ public boolean delete(@PathVariable("id") Long id) {
     @ApiPageable
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/controller/UserController.java]: Rostyk is going to remove this annotation. Please review his PR
https://github.com/IF-103-Java/ims/pull/170

And @ApiIgnore below will not be needed too.

File: src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -294,40 +284,40 @@ private PreparedStatement getPreparedStatement(User user, Connection connection)
                 SELECT *
                 FROM users
                 WHERE id = ?
+                AND active = 1
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: and by account_id

File: src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -294,40 +284,40 @@ private PreparedStatement getPreparedStatement(User user, Connection connection)
...
 
         public static final String SQL_SELECT_ADMIN_BY_ACCOUNT_ID = """
                 SELECT *
                 FROM users
                 WHERE role = 'ROLE_ADMIN'
                 AND account_id = ?
+                AND active = 1
             """;
 
         public static final String SQL_UPDATE_USER = """
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]:
```
public static final String SQL_SET_ACTIVE_STATUS_USER = """
                UPDATE users
                SET active = ?
                WHERE id = ?
            """;
```

Here also by account id and for hard delete as well

Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: I've secured these endpoints in another way

Comment (ikrasWork) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: Any troubles to update query?

Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: I've secured the delete method at the controller layer and here it is useless.

Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: What do you think about it?

Comment (ikrasWork) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: I saw, you approach. For this particular case it is not efficient.
For findById I agree, partially. But for soft delete not.

Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: Ok

Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java]: Thanks)

**Pull request merged**